### PR TITLE
#22 DRY up servo handling in StateManager

### PR DESCRIPTION
**Pull request by robberwick**
refactor out the servo initialising and manipulation logic to be a little DRYer.

### CODE REVIEW

File: libs/statemanager/src/statemanager.cpp [hunk 1]
```diff
@@ -5,10 +5,21 @@
...
+
 #include "motor2040.hpp"
 #include "servo.hpp"
 
 namespace STATEMANAGER {
+    void StateManager::initialiseServo(servo::Servo *servo, const uint pin, const float minPulse, const float midPulse, const float maxPulse, const float minValue = -0.7854, const float midValue = 0, const float maxValue = 0.7854) {
+        servo = new servo::Servo(pin);
+        servo->init();
+        servo->calibration().apply_three_pairs(minPulse, midPulse, maxPulse, minValue, midValue, maxValue);
+        servo->calibration().apply_three_pairs(1221, 1670, 2200, -0.7854, 0, 0.7854);
```
Comment (markmellors) [inline libs/statemanager/src/statemanager.cpp:19]: I think this line is there by mistake

**Review by markmellors — COMMENTED** [review]
generally looks good, I've not tested it yet though. I think line 19 needs removing otherwise the calibration will be wrong for one side, and that'd be difficult to see in a quick test

File: libs/statemanager/src/statemanager.cpp (hunk 1 above)
Comment (robberwick) [reply libs/statemanager/src/statemanager.cpp:19]: indeed it is! I'll yoink that _right out_

**Comment by markmellors** [comment]
tested - servos don't work. will investigate further

**Comment by markmellors** [comment]
further discoveries: 
`angle = driveTrainState.angles.right;` and `angle = driveTrainState.angles.left;`  are working correctly, both servos have the correct value here
The servos are never being disabled, even when the speed is less than <0.05. The         `if (std::fabs(speed) > 0.05) ` is working correctly (speed variable is correct), it's something to do with the servo object
The left servo is being initialised. If I change the last line of `initialiseServo` to  `servo->to_min();` or `servo->to_max();` , the left servo does as it should. The right servo appears to not be initialised/enabled at all.

**Comment by markmellors** [comment]
I figured it out. `initialiseServo` couldn't modify the pointer it was being passed, so it wasn't creating the servo objects as intended. The pointer is now passed as a reference, and it works :-)

**Comment by robberwick** [comment]
Oh that's a nice gotcha. Fine sleuthing, that man! 🔎🪲

**Comment by markmellors** [comment]
tested, works

**Pull request merged**